src/component/endorsements/endorsement.jsx

```diff
@@ -0,0 +1,222 @@
+import React, { useState, useEffect, useCallback } from "react";
+import {
+  Card,
+  Typography,
+  CardActions,
+  IconButton,
+  Snackbar,
+  CardContent,
+  Chip
+} from "@material-ui/core";
+import AddCircleIcon from "@material-ui/icons/AddCircle";
+import MuiAlert from "@material-ui/lab/Alert";
+import AddIcon from "@material-ui/icons/Add";
+
+import validate from "./yup";
+
+import { useStyles } from "./style";
+
+import { Modal } from "./features";
+
+import constant from "../../lib/constant";
+
+const Endorsement = () => {
+  const classes = useStyles();
+  const { snackBar, cardTitle, field } = constant;
+  const [open, setOpen] = useState(false);
+  const [openAlert, setOpenAlert] = useState(false);
+  const [value, setValue] = useState("");
+  const [endorsement, setEndorsement] = useState([]);
+  const [newArray, setNewArray] = useState([]);
+  const [message, setMessage] = useState("");
+  const [error, setError] = useState(false);
+  const [perPage] = useState(5);
+  const [initial, setInitial] = useState(0);
```

Comment:
useState hooks are feasible only in case when one has to manipulate  2-3 states only but when there are 4 or more than 4 states to handle in a single file it is better to use Class component